context.go

```diff
@@ -88,11 +96,14 @@ func (c SpanContext) IsDebug() bool {
 
 // IsValid indicates whether this context actually represents a valid trace.
 func (c SpanContext) IsValid() bool {
-	return c.traceID != 0 && c.spanID != 0
+	return c.traceID.IsValid() && c.spanID != 0
```

Comment:
Partially laziness - compiler doesn't complain about comparison with 0, but did complain for TraceID, which is a struct.

We do loop and ensure random_id() never returns 0s.